### DISCUSSION TITLE
FM broadcast interference

### DISCUSSION THREAD

**Discussion by chouyu** [egzumer/uv-k5-firmware-custom#212] [General]
A strange issue occurs when I use a non-resonant long antenna on the local ATC frequency of 125.2MHz. With the egzumer firmware, it introduces FM broadcast interference, making it unable to properly demodulate weak signals. However, when using the reppad firmware, it is not affected by the interference and can successfully monitor weak signals.

**Comment by g4eml** [comment]
Is the interference present all the time, or is it only there when a weak signal is present? 
If it is only there when a weak signal is present then it is probably intermodulation. 
The two firmwares may have different gain distribution which could cause this. Too much Pre-mixer gain could be the cause.

**Reply by egzumer** [reply]
> Too much Pre-mixer gain could be the cause.

This is really interesting. Are you saying that there is a difference on which block the gain is set if the value of the gain is the same? Is there anything I should pay attention to when applying gains?

I did change the gains table in AM-fix, maybe I should restructure it so it prefers one type of a gain over the other.

**Reply by g4eml** [reply]
I have occasionally noticed the intermodulation myself since the last gain table change. You can sometimes hear it as a faint voice superimposed on top of the signal you are listening to. Not bad, but certainly noticeable where it was not before. 

From what I can tell the LNA has two gain controls, one course and the other fine. Those will be working at the RF input frequency and should really be the first ones to be reduced in the event of strong signals. That will prevent the mixer being overloaded which is the cause of most large signal problems. The Mixer Gain will be at the IF frequency and is probably not that critical. The PGA gain will be after the mixer just before the IF is fed into the DACs and is probably the last one to be adjusted. Whilst the total gain is important the LNA gain probably has the most effect on interference.

I have just done a test with a modified gain table based on this theory and it seems to be working OK.  Rather than juggle with the various gains to get 1dB steps I concentrated on reducing the LNA gain first. Only when that was at the minimum did I start reducing the PGA and Mixer Gains. Unfortunately it is not possible to get 1dB increments by doing this, they are more like 3dB and the table is smaller, however on first testing it still seems to work fine. (and it saves some rom space!) .  I will do some more detailed tests on the bench tomorrow and will submit a PR if it looks OK. 

Colin.

**Reply by egzumer** [reply]
That is great. I think there is no need for 1dB steps. What you described seems more important.

**Comment by chouyu** [comment]
> Is the interference present all the time, or is it only there when a weak signal is present? If it is only there when a weak signal is present then it is probably intermodulation. The two firmwares may have different gain distribution which could cause this. Too much Pre-mixer gain could be the cause.

By using a long antenna, in theory, the signals from all frequencies should be stronger than those received with the original antenna. With the long antenna, I can receive weaker signals, but the strong signals from other frequencies are also amplified. I believe your theory is correct, and I hope to see results and corrections as soon as possible.

**Comment by ErikS-web** [comment]
Use an external band filter or make a band stop for the frequency that is bothering you.
